front/lib/resources/content_fragment_resource.ts

```diff
@@ -365,13 +367,69 @@ async function renderFromFileId(
   } else {
     const content = await getProcessedFileContent(workspace, fileId);
 
+    const messageWithFullContent = {
+      role: "content_fragment" as const,
+      name: `inject_${contentType}`,
+      content: [
+        {
+          type: "text",
+          text: `<attachment type="${contentType}" title="${title}">\n${content}\n</attachment>`,
+        } as Content,
+      ],
+    };
+
+    if (contentType !== "text/csv") {
+      return new Ok(messageWithFullContent);
+    }
+
+    // For CSV files we have a custom rule:
+    // We render the full content if the csv is small or a snippet if the csv is big.
+    // A big CSV is a CSV for which tokens counts for more than 10% of the context size of the model.
+    const tokenSize = await tokenCountForText(content, model);
```

Comment:
We likely don't want to count tokens on very large files? We can likely get away with an heuristic on file size.

More importantly we don't want to download the CSV at all here if it's large?

Reply:
I'll use the `fileSize` of the file or `textBytes` of the contentFragment to estimate the size 👍🏻